src/scanner.js

```diff
@@ -19,50 +19,42 @@ async function scanner(cfg, authorizer) {
   // for scans to take longer than for the auth service to be overloaded.
   let auth = new taskcluster.Auth({credentials: cfg.app.credentials});
 
-  // gather all identityProviderIds used in any authorizer
-  let identityProviders = authorizer.identityProviders;
+  const scan = async h => {
+    const handler = handlers[h];
+    const clients = await auth.listClients({prefix: handler.identityPrefix});
 
-  // enumerate all clients for any of those identity providers
-  let clients = [];
-  for (let idp of identityProviders) {
-    clients = clients.concat(await auth.listClients({prefix: idp + '/'}));
-  }
+    // iterate through the clients, constructing a new User as necessary, comparing
+    // the client's scopes to the User's scopes and disabling where necessary.
+    let user, userScopes;
 
-  // sort by clientId, so that each identity (a prefix of the clientId) appears
-  // contiguously
-  clients = _.sortBy(clients, 'clientId');
+    for (let client of clients) {
+      debug('examining client', client.clientId);
+      if (!client.clientId.match(CLIENT_ID_PATTERN) || client.disabled) {
+        continue;
+      }
 
-  // iterate through the clients, constructing a new User as necessary, comparing
-  // the client's scopes to the User's scopes and disabling where necessary.
-  let user, userScopes;
-  let idPattern = /^([^\/]*\/[^\/]*)\/.+$/;
-  for (let client of clients.sort()) {
-    debug('examining client', client.clientId);
-    if (!client.clientId.match(idPattern) || client.disabled) {
-      continue;
-    }
-
-    // refresh the user if it does not correspond to this client
-    let clientIdentity = client.clientId.replace(idPattern, '$1');
-    if (!user || user.identity != clientIdentity) {
-      user = new User();
-      user.identity = clientIdentity;
+      // when client has a github login, `patternMatch` will have an extra index entry with the user's GH username
+      // e.g., ['mozilla-auth0/github|0000/helfi92, 'mozilla-auth0/github|0000', 'helfi92']
```

Comment:
Could we factor this so that "/helfi92" is totally ignored here?

Reply:
I'm not sure I understand what you mean here but what I can do is have a `identityFromClientId` method in mozilla-auth0. This will abstract the complexity away from the scanner file.